x-pack/plugins/reporting/server/lib/screenshots/observable.ts

```diff
@@ -46,8 +47,8 @@ export function getScreenshots$(
   browserDriverFactory: HeadlessChromiumDriverFactory,
   opts: ScreenshotObservableOpts
 ): Rx.Observable<ScreenshotResults[]> {
-  const apmTrans = apm.startTransaction(`reporting screenshot pipeline`, 'reporting');
-  const apmCreatePage = apmTrans?.startSpan('create_page', 'wait');
+  const apmTrans = apm.startTransaction('screenshot-pipeline', REPORTING_TRANSACTION_TYPE);
```

Comment:
For reporting, the goal is to have the data look like this:

| transaction.type  | transaction.name |
| ------------- | ------------- | 
| reporting  | execute-job-pdf-v2  | 
| reporting  | execute-job-pdf  | 
| reporting  | generate-pdf  | 

I guess we can't add the name of the report for privacy reasons, but I think we could save the ID of the dashboard - so we could pass it to the customer in case we see a problem with one of the reports.

Reply:
Rather than using an ID of a Dashboard, report job parameters use a LocatorID and LocatorState object, so it can be a deep link to any app registered with the Locator Service. I'm not sure APM data is the best place to keep the link state.

What you bring up touches on the subject of Monitoring in Reporting. I'd love to get more collaboration on a document I'm working on for this: https://docs.google.com/document/d/1eDi74wTQe0vTZ6HT-_7O9fggckrvDpIXqIrTvNTov44/edit?usp=sharing - please take a look :)